assisted-events-scrape/events_scrape/events_scrape.py

```diff
@@ -38,225 +31,73 @@
+        self.max_idle_minutes = config["max_idle_minutes"]
+        self.changes = Changes()
+        self.error_counter = ErrorCounter()
 
-def get_no_name_message(event_message: str, event_names: list):
-    event_message = re.sub(r"^Host \S+:", "", event_message)
-    for name in event_names:
-        event_message = event_message.replace(name, "Name")
-    event_message = re.sub(UUID_REGEX, "UUID", event_message)
-    return event_message
+    def start_workers(self, config: dict) -> None:
 
+        self.clusters_queue = Queue()
+        workers_config = {
+            "queue": self.clusters_queue,
+            "error_counter": self.error_counter,
+            "changes": self.changes,
+            "sentry": config["sentry"]
+        }
+        for n in range(config["n_workers"]):
+            workers_config["name"] = f"Worker-{n}"
+            worker = ClusterEventsWorker(workers_config,
+                                         cluster_repository=self.cluster_repo,
+                                         event_repository=self.event_repo,
+                                         cluster_events_storage=self.cluster_events_storage)
 
-def get_cluster_object_names(cluster_bash_data):
-    strings_to_remove = list()
-    for host in cluster_bash_data["cluster"]["hosts"]:
-        host_name = host.get("requested_hostname", None)
-        if host_name:
-            strings_to_remove.append(host_name)
-    strings_to_remove.append(cluster_bash_data["cluster"]["name"])
-    return strings_to_remove
+            worker.daemon = True
+            worker.start()
 
+    def is_idle(self):
+        return not self.changes.has_changed_in_last_minutes(self.max_idle_minutes)
 
-def process_metadata(metadata_json):
-    p = process.GetProcessedMetadataJson(metadata_json)
-    return p.get_processed_json()
+    def has_too_many_unexpected_errors(self):
+        return self.error_counter.get_errors() > self.errors_before_restart
 
+    def run_service(self):
 
-def get_doc_id(event_json):
-    id_str = event_json["event_time"] + event_json["cluster_id"] + event_json["message"]
-    _id = int(hashlib.md5(id_str.encode('utf-8')).hexdigest(), 16)
-    return str(_id)
+        clusters = self.cluster_repo.list_clusters()
 
+        if not clusters:
+            log.warning(f'No clusters were found, waiting {RETRY_INTERVAL / 60} min')
+            time.sleep(RETRY_INTERVAL)
+            return None
 
-def process_event_doc(event_data, cluster_bash_data):
-    cluster_bash_data.update(event_data)
+        cluster_count = len(clusters)
+        for cluster in clusters:
+            self.clusters_queue.put(cluster)
+        log.info(f"Added {cluster_count} cluster IDs to queue, joining queue...")
+        self.clusters_queue.join()
+        log.info("Finish syncing all clusters - sleeping 30 seconds")
+        time.sleep(30)
 
 
 def get_env(key, mandatory=False, default=None):
     res = os.environ.get(key, default)
```

Comment:
when res would be empty?

Reply:
if you will change the configuration to be a data class it won't be necessary. right?